pkg/mobile/bindableServices.go

```diff
@@ -0,0 +1,189 @@
+package mobile
+
+import (
+	"encoding/json"
+	"strings"
+
+	scv1beta1 "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1beta1"
+	k8v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type BindableMobileServiceCRUDLImpl struct {
+	scClient         scv1beta1.ServicecatalogV1beta1Interface
+	mobileClientRepo MobileClientRepo
+	secretsCRUDL     SecretsCRUDL
+}
+
+func NewServiceBindingLister(scClient scv1beta1.ServicecatalogV1beta1Interface, mobileClientRepo MobileClientRepo, secretsCRUDL SecretsCRUDL) *BindableMobileServiceCRUDLImpl {
+	return &BindableMobileServiceCRUDLImpl{
+		scClient:         scClient,
+		mobileClientRepo: mobileClientRepo,
+		secretsCRUDL:     secretsCRUDL,
+	}
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) Create(namespace string, binding *ServiceBinding, formData map[string]interface{}) (*ServiceBinding, error) {
+
+	bindingsApi := lister.scClient.ServiceBindings(namespace)
+	binding2, err := bindingsApi.Create(binding)
+	if err != nil {
+		return nil, err
+	}
+
+	parametersSecret := makeParametersSecret(binding, binding2, formData)
+	_, err = lister.secretsCRUDL.Create(namespace, parametersSecret)
+
+	if err != nil {
+		return nil, err
+	}
+	return binding2, nil
+
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) Delete(namespace string, bindingName string) error {
+	return lister.scClient.ServiceBindings(namespace).Delete(bindingName, &v1.DeleteOptions{})
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) List(namespace string, mobileClientName string) (*BindableMobileServiceList, error) {
+
+	listOpts := v1.ListOptions{}
+
+	getOpts := v1.GetOptions{}
+
+	serviceBindingList := BindableMobileServiceList{}
+
+	serviceBindings, err := lister.scClient.ServiceBindings(namespace).List(listOpts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	serviceInstances, err := lister.scClient.ServiceInstances(namespace).List(listOpts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, serviceInstance := range serviceInstances.Items {
+
+		bindableService := BindableMobileService{}
+
+		csc, err := lister.scClient.ClusterServiceClasses().Get(serviceInstance.Spec.ClusterServiceClassRef.Name, getOpts)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if csc.Spec.ExternalMetadata != nil {
```

Comment:
might be better if we move this into its own function to reduce the size of this one.